### PR TITLE
Standalone Docker image

### PR DESCRIPTION
**Pull request by tombh**
This builds on @JonDum's fork (mentioned in #45) to create a Docker image that automatically adds all queues prefixed with 'bull' or whatever value is specified in the `BULL_PREFIX` ENV var.

I've included it in the README, but I'll repeat the command here with my username as it's a nice oneliner to quickly try out and see what this PR is introducing: `docker run --net=host --rm tombh/bull-board`

I've also taken the liberty of including an update to this repo's Github Actions. It doesn't currently work as it needs someone (I'm assuming @vcapretz) to create a Docker Hub account and create a registry token. Once signed up that can be done here: https://hub.docker.com/settings/security

### CODE REVIEW

**Comment by tombh** [comment]
I think there's a problem with the `refreshQueues()` function generating new Redis client connections every time. The number of client connections reported in the UI seems to grow and grow. Or that could just be some problem with the way I'm using Bull queues in my actual app...

**Comment by tombh** [comment]
Thanks for the original code @JonDum!

I can't think of a case where 
```js
let queue_names = []
...
if (!queue_names.includes(queue_name)) {
  ...
  queue_names.push(queue_name)
}
```
isn't strong enough.

Of course it is possible for a plain JS array to have duplicates, but there is nowhere in the code where a duplicate could be inserted, thus it's safe?

**Comment by JonDum** [comment]
You're right there, should be good on duplicates. What if someone deletes/removes queues though? I think in my app some queues are transient so I had it done that way so they would be cleared out when they were no longer in redis.

**Comment by tombh** [comment]
Ah, I hadn't thought of that, so yeah we definitely need a way to manage duplicates then.

So the thing is that Bull Board's `setQueues()` doesn't seem to sync state, it just adds any new queues it's given - therefore it doesn't delete queues it already has but isn't given. I haven't dug into Bull Board's code to look into it, maybe someone more familiar could comment on that behaviour?

Another thing to bear in mind is that the reason I changed your `refreshQueues()` is that it was leaking Redis connections. So it'll still need a little tweak anyway to only issue `new Queue()` once for each new queue.

**Comment by thelinuxlich** [comment]
I tried to run this image but it is complaining about password(which I don't want to setup any)

File: standalone.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const { setQueues, router } = require('./dist/index')
+const Queue = require('bull')
+const redis = require('redis')
+const app = require('express')()
+
+const redisOptions = {
+  redis: {
+    port: process.env.REDIS_PORT || 6379,
+    host: process.env.REDIS_HOST || 'localhost',
+    password: process.env.REDIS_PASSWORD || '',
```
Comment (srosset81) [inline standalone.js:10]: This doesn't work if no password is set, it should be
```suggestion
    password: process.env.REDIS_PASSWORD || undefined,
```

**Comment by tombh** [comment]
Thanks @thelinuxlich and @srosset81. I fixed it. And rebased against latest master. And pushed and built the latest image to Docker Hub as `tombh/bull-board`, same image name as before.

**Comment by tombh** [comment]
Is the delay to merging this the failing CI?

**Comment by tombh** [comment]
I fixed the package.json conflict and commented out the failing line in the build step to deploy the Docker image to the Docker Hub. I can't see the failure in the snyk build step as it's behind a login.

File: standalone.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const { setQueues, router } = require('./dist/index')
```
Comment (vcapretz) [inline standalone.js:1]: could you convert this file to `.ts` and remove the npm lock file too, please?

**Comment by airhorns** [comment]
Would be super awesome to get this in @tombh !

**Comment by airhorns** [comment]
Ah -- one thing this is missing would be detection of `bullmq` queues vs plain ole `bull` queues, and the wrapping in the appropriate adapter. Is anyone aware of a way to infer which kind of queue it is by looking at redis?

**Comment by tombh** [comment]
I'm definitely planning to sort this out, just life and stuff, etc, you know how it is...

**Comment by erwineverts** [comment]
I have created a separate repository for a docker image: https://gitlab.com/erwineverts/bull-board-docker

**Comment by henrysachs** [comment]
any possibility getting back some track on this? Would love this. I currently try to integrate that into nest with nx but its compiling all sources to a single main.js. thats why all references to files dont work.

**Comment by felixmosh** [comment]
> any possibility getting back some track on this? Would love this. I currently try to integrate that into nest with nx but its compiling all sources to a single main.js. thats why all references to files dont work.

I'm not sure I got the issue.

This issue is too old, what version do you use?

**Comment by henrysachs** [comment]
> 

Hey there I tried to explain my Issue a bit more in detail in: #347

**Comment by felixmosh** [comment]
Closing since it is out of sync with master branch

**Pull request closed without merging**